internal/proxy/powerscale_handler.go

```diff
@@ -0,0 +1,454 @@
+		if err != nil {
+			writeError(w, "powerscale", "failed to parse remote host", http.StatusInternalServerError)
+			return
+		}
+		s.log.Printf("RemoteAddr: %s", host)
+
+		pvName := r.Header.Get(HeaderPVName)
+		log.Printf("pvName: %s", pvName)
+		// Update metrics counter for volumes requested.
+		//volReqCount.Add(pvName, 1)
+
+		// Ask OPA to make a decision
+
+		jwtGroup := r.Context().Value(web.JWTTenantName)
+		group, ok := jwtGroup.(string)
+		if !ok {
+			writeError(w, "powerscale", "incorrect type for JWT group", http.StatusInternalServerError)
+			return
+		}
+
+		jwtValue := r.Context().Value(web.JWTKey)
+		jwtToken, ok := jwtValue.(token.Token)
+		if !ok {
+			writeError(w, "powerscale", "incorrect type for JWT token", http.StatusInternalServerError)
+			return
+		}
+
+		claims, err := jwtToken.Claims()
+		if err != nil {
+			writeError(w, "powerscale", "decoding token claims", http.StatusInternalServerError)
+			return
+		}
+
+		s.log.Println("Asking OPA...")
+		// Request policy decision from OPA
+		// The driver does not send the volume request size so we set the volumeSizeInKb to 0
+		ans, err := decision.Can(func() decision.Query {
+			return decision.Query{
+				Host:   opaHost,
+				Policy: "/karavi/volumes/powerscale/create",
+				Input: map[string]interface{}{
+					"claims":          claims,
+					"request":         map[string]interface{}{"volumeSizeInKb": 0},
+					"storagepool":     isiPath,
+					"storagesystemid": systemID,
+					"systemtype":      "powerscale",
+				},
+			}
+		})
+		var opaResp CreateOPAResponse
+		err = json.NewDecoder(bytes.NewReader(ans)).Decode(&opaResp)
+		if err != nil {
+			s.log.Printf("decoding opa response: %+v", err)
+			writeError(w, "powerscale", "decoding opa request body", http.StatusInternalServerError)
+			return
+		}
+		log.Printf("OPA Response: %+v", opaResp)
+		if resp := opaResp.Result; !resp.Allow {
+			reason := strings.Join(opaResp.Result.Deny, ",")
+			s.log.Printf("request denied: %v", reason)
+			writeError(w, "powerscale", fmt.Sprintf("request denied: %v", reason), http.StatusBadRequest)
+			return
+		}
+
+		// At this point, the request has been approved.
+		// The driver does not send the volume request size so we set the Capacity to 0 to always approve the quota
+		qr := quota.Request{
+			SystemType:    "powerscale",
+			SystemID:      systemID,
+			StoragePoolID: isiPath,
+			Group:         group,
+			VolumeName:    pvName,
+			Capacity:      "0",
+		}
+
+		// Reset the original request
+		err = r.Body.Close()
+		if err != nil {
+			s.log.Printf("Failed to close original request body: %v", err)
+		}
+		r.Body = ioutil.NopCloser(bytes.NewBuffer(b))
+		sw := &web.StatusWriter{
+			ResponseWriter: w,
+		}
+
+		s.log.Println("Proxying request...")
+		// Proxy the request to the backend powerflex.
+		r = r.WithContext(ctx)
+		next.ServeHTTP(sw, r)
+
+		// TODO(aaron): Determine if when the approved volume fails the volume is
```

Comment:
Will this be addressed in this PR?

Reply:
This comment was copied from the powerflex handler. I think it can be removed.

Reply:
Removed.